### PR TITLE
Make interpolate_bilinear compatible with TF Dataset (#332)

### PR DESCRIPTION
**Pull request by mels630**
* Make interpolate_bilinear compatible with TF Dataset

* Split InterpolateBilinearTest from DenseImageWarpTest

* Add InterpolateBilinearTest::test_unknown_shape

### CODE REVIEW

**Comment by seanpmorgan** [comment]
Hi @mels630 , could you run the below to auto-format the code:
`bash tools/run_docker.sh -c 'make code-format'`

File: tensorflow_addons/image/dense_image_warp.py [hunk 1]
```diff
@@ -48,62 +49,76 @@ def interpolate_bilinear(grid, query_points, indexing="ij", name=None):
...
+                    tf.debugging.assert_less_equal(
+                        tf.cast(
+                            grid_shape[0] * grid_shape[1] * grid_shape[2],
+                            dtype=tf.dtypes.float32),
+                        np.iinfo(np.int32).max / 8.0,
+                        message="The image size or batch size is sufficiently "
+                        "large that the linearized addresses used by "
+                        "tf.gather may exceed the int32 limit.")
+            ]):
+                # pylint: enable=bad-continuation
```
Comment (WindQAQ) [inline tensorflow_addons/image/dense_image_warp.py]:
```suggestion
            # pylint: enable=bad-continuation
```

File: tensorflow_addons/image/dense_image_warp.py [hunk 2]
```diff
@@ -48,62 +49,76 @@ def interpolate_bilinear(grid, query_points, indexing="ij", name=None):
...
+        query_static_shape = query_points.shape
         query_shape = tf.shape(query_points)
+        if query_static_shape.dims is not None:
+            if len(query_static_shape) != 3:
+                raise ValueError("Query points must be 3 dimensional.")
+            query_hw = query_static_shape[2]
+            if query_hw is not None and query_hw != 2:
+                raise ValueError("Query points last dimension must be 2.")
+        else:
+            with tf.control_dependencies([
```
Comment (WindQAQ) [inline tensorflow_addons/image/dense_image_warp.py]:
```suggestion
            # pylint: disable=bad-continuation
            with tf.control_dependencies([
                    tf.debugging.assert_equal(
                        query_shape[2],
                        2,
                        message="Query points last dimension must be 2.")
            ]):
            # pylint: enable=bad-continuation
```

**Comment by mels630** [comment]
Thanks, @WindQAQ. Let me know if local policy is such that I should squash the commits.

File: tensorflow_addons/image/dense_image_warp.py [hunk 3]
```diff
@@ -48,62 +49,83 @@ def interpolate_bilinear(grid, query_points, indexing="ij", name=None):
...
+                        query_shape[2],
+                        2,
+                        message="Query points last dimension must be 2.")
+            ]):
+            # pylint: enable=bad-continuation
+                    tf.debugging.assert_equal(
+                        query_shape[2],
+                        2,
+                        message="Query points last dimension must be 2.")
+            ]):
```
Comment (WindQAQ) [inline tensorflow_addons/image/dense_image_warp.py]: line 103 to line 107 are redundant.

Comment (mels630) [reply tensorflow_addons/image/dense_image_warp.py]: thanks. i failed to update things properly using the web tools. should be better now.

**Pull request merged**